### PR TITLE
Implement rfc6587 auto detection

### PR DESCRIPTION
**Pull request by bazsi**
This is a pretty common misconfiguration that should be handled automatically. 

Just use syslog(transport(auto)) to automatically accept both octet counted and non-octet counted connections, this opens a TCP port without TLS, meaning it is an implementation of RFC6587.

### CODE REVIEW

File: lib/logproto/logproto-auto-server.c [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  else if (rc < 0)
+    return LPS_ERROR;
+
+  self->proto_impl = _construct_detected_proto(self, detect_buffer, rc);
+  if (self->proto_impl)
+    {
+      /* transport is handed over to the new proto */
+      self->super.transport = NULL;
+      return LPS_SUCCESS;
+    }
```
Comment (alltilla) [inline lib/logproto/logproto-auto-server.c:121]: Shouldn't we do a handshake after we autodetected the proto?

File: news/feature-4814.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+`syslog()` source driver: add support for RFC6587 style auto-detection of
+octet-count based framing to avoid confusion that stems from the sender
+using a different protocol to the server.  This is automatically enabled for
+both the `transport(tcp)` and the `transport(tls)` settings.  You can disable
+auto-detection by using `transport(text)` or `transport(framed)`.
```
Comment (alltilla) [inline news/feature-4814.md:5]: We disable auto detection with `transport(force-framed)`, right?

**Comment by bazsi** [comment]
This should wait until #156 is merged.

**Comment by bazsi** [comment]
I have now rebased it but still need to check if it works on top of #156

**Comment by alltilla** [comment]
#156 is now merged, I think we can give this another go (rebase, CI, review, ...)

**Comment by bazsi** [comment]
This should now be ready for review.

Note: there's a followup PR as well, #361

**Comment by bazsi** [comment]
I have now opted out of doing this by default. The auto-detection mechanism is only enabled with transport(auto)

**Comment by MrAnno** [comment]
I've started reviewing this patchset, will finish it by the end of the week.

**Review by OverOrion — COMMENTED** [review]
I've reviewed this and tested it manually as well, worked just fine :rocket:.
Is there a reason why it's still marked as `WIP`?

**Comment by bazsi** [comment]
I pulled some of the later commits into this one, rebased against main and removed WIP.

@OverOrion if you can please re-approve, but note that I did pull in some newer changes into this.

**Pull request merged**